icu4c/source/test/intltest/erarulestest.cpp

```diff
@@ -0,0 +1,130 @@
+// © 2018 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+#include "unicode/utypes.h"
+
+#if !UCONFIG_NO_FORMATTING
+
+#include "unicode/calendar.h"
+#include "unicode/localpointer.h"
+#include "unicode/unistr.h"
+#include "erarules.h"
+#include "erarulestest.h"
+
+void EraRulesTest::runIndexedTest(int32_t index, UBool exec, const char* &name, char* /*par*/)
+{
+    if (exec) {
+        logln("TestSuite EraRulesTest");
+    }
+    TESTCASE_AUTO_BEGIN;
+    TESTCASE_AUTO(testAPIs);
+    TESTCASE_AUTO(testJapanese);
+    TESTCASE_AUTO_END;
+}
+
+void EraRulesTest::testAPIs() {
+    const char * calTypes[] = {
```

Comment:
no other way to iterate on these types?

Reply:
ICU4J has CalType enum. ICU4C has calendar type enum, but their names are defined in calendar.cpp (https://github.com/unicode-org/icu/blob/master/icu4c/source/i18n/calendar.cpp#L159) and has no API exposed to get calendar type name from enum or vise versa.

Reply:
Filed https://unicode-org.atlassian.net/projects/ICU/issues/ICU-20134